check_samples_double_format.sh

```diff
@@ -8,5 +8,10 @@ for file in "$@"; do
     ./grol $file > /tmp/output1
     ./grol /tmp/format2 > /tmp/output2
     diff -u /tmp/output1 /tmp/output2
+    ./grol -format -compact $file > /tmp/format3
+    ./grol -format -compact /tmp/format3 > /tmp/format4
+    diff -u /tmp/format3 /tmp/format4
+    ./grol /tmp/format4 > /tmp/output3
+    diff -u /tmp/output1 /tmp/output3
```

Comment:
this is the test, makes sure the compact form is stable and that it evaluates to same as non compact

probably could have a few unit test too

Reply:
added unit test, not 100% sure there isn't a corner case but ... hopefully bug would get reported